Add onStats message

gpongelli opened the pull request:
changes to handle onStats mqtt message 
`[deebot_client.mqtt_client] Got message: topic=iot/atr/onStats/f266fb63-5747-4cca-b8fb-3c1ceda9bd3f/659yh8/r7j2/j, payload=b'{"header":{"pri":1,"tzm":480,"ts":"1745398159530","ver":"0.0.1","fwVer":"1.4.9","hwVer":"0.1.1"},"body":{"data":{"area":27,"time":2459,"cid":"105534963","start":"1745395672","type":"auto","enablePowerMop":1,"powerMopType":2,"aiopen":1,"aitypes":[9],"avoidCount":1}}}'`

please let me know how can I test the changes on my deebot T9 AIVI, thanks.

closes #592

gpongelli committed:
move to correct device

gpongelli committed:
manage mqtt message onStats

edenhaus commented on deebot_client/commands/json/stats.py:
```diff
@@ -51,3 +52,14 @@ def _handle_body_data_dict(
+class GetOnStats(OnStats, JsonCommandWithMessageHandling):
+    """Get onStats command."""
+
+    NAME = "getOnStats"
```
Are you sure this is a valid command? Did you test your code?

gpongelli replied:
I didn’t test this code, I’ve no idea on how to test it on my T9.
Is there a way to gather the command list directly from the device ? 

anyway, I added it to have a command into device’s capability, otherwise I’ve no idea on how to “link” mqtt data to the device’s class, can you point me on how to do that?

edenhaus replied:
> Is there a way to gather the command list directly from the device ?

Unfortunately not. We can only use trial and error.
You can checkout the example of the readme and test if you get an answer from your bot to test the command

gpongelli replied:
I've removed this command, because those stats come from mqtt topic "iot/atr/onStats/" and I've seen device subscribes to topics; let me know if commands are needed to handle mqtt messages.

edenhaus commented on deebot_client/commands/json/stats.py:
```diff
@@ -51,3 +52,14 @@ def _handle_body_data_dict(
+
+    NAME = "getOnStats"
+
+    def __init__(self, *, is_available_check: bool = False) -> None:
+        super().__init__()
+        self._is_available_check = is_available_check
```
```suggestion
```

edenhaus commented on deebot_client/messages/json/__init__.py:
```diff
@@ -17,6 +17,7 @@
     "OnBattery",
     "OnMapSetV2",
     "ReportStats",
+    "OnStats",
```
Please sort it

gpongelli replied:
done

edenhaus commented on deebot_client/messages/json/__init__.py:
```diff
@@ -30,6 +31,8 @@
     OnStationState,
 
+    OnStats,
+
```
```suggestion
```

edenhaus reviewed (changes requested):
Please add tests

gpongelli committed:
Suggested change deebot_client/messages/json/__init__.py

Co-authored-by: Robert Resch <robert@resch.dev>

gpongelli committed:
Commit suggestion on deebot_client/commands/json/stats.py

Co-authored-by: Robert Resch <robert@resch.dev>

gpongelli committed:
sort list

gpongelli committed:
remove GetOnStats, statistics are managed via mqtt

gpongelli committed:
test for OnStats

gpongelli committed:
add test_onStats

test data comes from logs on gh issue; other fields are filtered out
for now to not change StatsEvent dataclass.

gpongelli commented:
> Please add tests

added `test_onStats` ; let me know if commands is needed to handle mqtt data, thanks.

gpongelli committed:
revert change

edenhaus commented:
pre-commit.ci autofix

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

gpongelli committed:
Merge branch 'dev' into deebot-t9-mqtt-stats

gpongelli commented:
pre-commit.ci autofix

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

edenhaus committed:
Revert capabilities change

edenhaus committed:
Fix test

edenhaus reviewed (approved):
Thanks @gpongelli :+1: 

I reverted the capabilities file changes as if they are still needed should go into a new PR
